File: solutions/typescript/square-root/1/square-root.ts
export function squareRoot(radicand: number): number {
    if (radicand === 0) return 0;
    let x = radicand;
    let y = (x + 1) / 2;
    while (y < x) {
        x = y;
        y = (x + radicand / x) / 2;
    }
    return x;
}

File: solutions/go/square-root/1/square_root.go
package squareroot

import (
	"errors"
	"math"
)

func SquareRoot(number int) (int, error) {
	if number < 0 {
		return 0, errors.New("negative number")
	}
	return int(math.Sqrt(float64(number))), nil
}
